recipes/csvmonkey/all/conanfile.py

```diff
@@ -0,0 +1,50 @@
+import os
+
+from conans.errors import ConanInvalidConfiguration
+from conans import ConanFile, CMake, tools
+
+required_conan_version = ">=1.36.0"
+
+class CSVMONEKYConan(ConanFile):
+    name = "csvmonkey"
+    license = "BSD-3-Clause"
+    url = "https://github.com/conan-io/conan-center-index"
+    description = "Header-only vectorized, lazy-decoding, zero-copy CSV file parser "
+    topics = ("csv-parser", "header-only", "vectorized")
+    homepage = "https://github.com/dw/csvmonkey/"
+    settings = "arch", "compiler"
+    no_copy_source = True
+    options = {"with_spirit": [True, False]}
+    default_options = {"with_spirit": False}
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    def validate(self):
+        if self.settings.arch not in ("x86", "x86_64",):
+            raise ConanInvalidConfiguration("{} requires x86 architecture.".format(self.name))
+
+        if self.settings.compiler == "Visual Studio":
+            raise ConanInvalidConfiguration("{} doesn't support Visual Studio C++.".format(self.name))
+
+    def requirements(self):
+        if self.options.with_spirit:
+            self.requires("boost/1.77.0")
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version], destination=self._source_subfolder, strip_root=True)
+
+    def package(self):
+        self.copy("LICENSE*", "licenses", self._source_subfolder)
+        self.copy("*.hpp", dst="include", src=os.path.join(self._source_subfolder, "include"))
+
+    def package_id(self):
+        self.info.header_only()
+
+    def package_info(self):
+        self.cpp_info.set_property("cmake_file_name", "csvmonkey")
+        self.cpp_info.set_property("cmake_target_name", "csvmonkey")
+        self.cpp_info.set_property("pkg_config_name", "csvmonkey")
```

Comment:
upstream doesn't provide pkgconfig or cmake files, it would be better to remove these lines, it's misleading.